code/game/objects/structures/stool_bed_chair_nest/chairs.dm

```diff
@@ -232,9 +232,23 @@
 	sheet_amt = 3
 	anchored = 1
 	overrideghostspin = 1
+	var/image/legs
+	var/couch_colour = "#37FDFC"  // currently cyan for testing
+	var/colourName = "grey" // dont think ill need this
 
 // layer stuff
 
+/obj/structure/bed/chair/comfy/couch/New()
+
+	legs = image("icons/obj/objects.dmi", "[icon_state]_legs", MOB_LAYER - 0.1)		// since i dont want the legs colored they are a separate overlay
+	legs.plane = PLANE_MOB
+	overlays += legs
+	..()
+	if(couch_colour)  // the couch color will have null as default
+		buckle_overlay.icon += couch_colour // doesnt work (why?)
```

Comment:
You're adding a colour string to a file - why on earth would it work?
You can set the colour directly on the object by doing src.color or buckle_overlay.color = couch_colour